lib/Doctrine/DBAL/Driver/PDOException.php

```diff
@@ -43,6 +44,13 @@ public function __construct(\PDOException $exception)
      */
     public function getErrorCode()
     {
+        /** @psalm-suppress ImpureMethodCall */
+        Deprecation::trigger(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/4112',
+            'Driver\AbstractException::getErrorCode() is deprecated, use getSQLState() or getCode() instead.'
```

Comment:
With the `__METHOD__` approach, this would be a fully qualified class name. Otherwise, it's inconsistent with the rest of the deprecation messages which mostly contain a class name without the namespace.